### DISCUSSION TITLE
Type autocompletions via JSDoc

### DISCUSSION THREAD

**Discussion by binarious** [openhab/openhab-js#255] [JS Maintainers Discussions]
I tried to get autocompletions via JSDoc type with `typescript-language-server` to work without success:

```js
const { Item } = require('openhab');

// ...

/** @param {Item} item */
function queryState(item) {
  const host = item.getMetadata('host').value;
}
```
In this example `item` doesn't get autocompletion, but `items.getItem('...').getMe...` does. I also tried `@param {items.Item} item` or `@param {typeof items.Item} item` without success. 

Any ideas?

**Comment by florian-h05** [comment]
Your import of `Item` is wrong and won't work.
Always have a look at the docs, they will tell you how to import correctly: https://github.com/openhab/openhab-js#standard-library.

This code will work:

```javascript
const { Item } = require('openhab').items;

/**
 * 
 * @param {Item} item 
 */
function queryState (item) {
  const host = item.getMetadata('host').value;
}
```

**Reply by binarious** [reply]
Thanks, this actually works. I was expected `const { items } = require('openhab');` and  `@param {items.Item} item` to also work, but it doesn't seem to be the case.

**Reply by florian-h05** [reply]
I am not sure why `items.Item` doesn’t work. You could use a `import(‚openhab/items/items‘).Item` inside the JSDoc type brackets.
